### PR TITLE
feat: ei-2934 - updating client library for latest API changes; 

### PR DESCRIPTION
**Pull request by conbon**
* Update the submodule to `v1.2.0` tag  of the IOTICS API and updates the sparql example to work with the new format.

* Update buf and make use of the remote plugins - `buf config migrate` was used and checked - import paths/exmaple code functionality was not impacted by this upgrade.
  * There is no longer a need for a heavyweight dockerfile/build due to using remote buf plugins (instead of building locally from source).

This has been tested against various environments using the setup described in the README.
Latest .env file used (has also been tested against alpha for the non-fkg examples):
```
SPACE=slo1.dev.iotics.space
USER_DID=did:iotics:iotNibth5ayhABaVk6hiVqPtXUgwj9XuvHwS
AGENT_DID=did:iotics:iotXxzrdHp937cak776CM82Mx3c2rwUDXehB
AGENT_KEY_NAME=#MyAgentKey
AGENT_NAME=#MyAgentName
AGENT_SECRET=xxx # can be provided if needed
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,25 @@
 # Iotics gRPC Python Client
-A Python library for interacting with Iotics API.
```
Comment (conbon) [inline README.md:2]: Mostly format changes to satisfy linter in IDE

Comment (conbon) [reply README.md:2]: plus removal of docker sections

File: src/iotics/api/host_pb2.pyi [hunk 2]
```diff
@@ -5,20 +5,16 @@ Copyright (c) 2019-2022 Iotic Labs Ltd. All rights reserved.
...
 import builtins
 import google.protobuf.descriptor
 import google.protobuf.message
 import iotics.api.common_pb2
-import sys
-
-if sys.version_info >= (3, 8):
-    import typing as typing_extensions
-else:
-    import typing_extensions
```
Comment (andyantrim) [inline src/iotics/api/host_pb2.pyi:17]: This means dropping support for 3.7, are we okay with that/is that already the case?

**Review by andyantrim — APPROVED** [review]
Looks good, just double check the python version compitablity. I'm pretty sure that it's already the case we don't support 3.7, maybe we should call out a min version in the readme somewhere?

File: src/iotics/api/host_pb2.pyi (hunk 2 above)
Comment (Elleppi) [reply src/iotics/api/host_pb2.pyi:17]: I've checked our internal and external users' Connectors in Python. They all use Python >= 3.8, so we can go ahead and remove support for 3.7

**Pull request merged**